pyproject.toml

```diff
@@ -0,0 +1,51 @@
+[build-system]
+requires = ["hatchling", "hatch-requirements-txt", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "mkdocs-material"
+dynamic = [
+  "version",
+  "dependencies",
```

Comment:
@ofek What does it mean for the dependencies to be dynamic?

`mkdocs-material` needs to depend on `mkdocs>=1.4.0`.

Reply:
@Kludex I can answer in part. Any field in pyproject.toml (besides `name`) can be dynamic, and it simply means that the build backend (e.g. hatchling) will provide this metadata: https://peps.python.org/pep-0621/#dynamic. My understanding is that the `dynamic` field itself is not directly used by PEP 517 frontends, rather it is useful for tools to know whether to ask the backend for metadata rather than just using the pyproject.toml file. @ofek might correct me here, though.

Reply:
The `pyproject.toml` file has `dependencies` set to `dynamic`, because the `requirements_txt` hook populates the `depedencies` metadata using https://github.com/squidfunk/mkdocs-material/blob/master/requirements.txt